src/protocol/http/src/main/java/org/apache/jmeter/protocol/http/visualizers/RequestViewHTTP.java

```diff
@@ -316,9 +322,10 @@ public static Map<String, String[]> getQueryMap(String query) {
                 System.arraycopy(known, 0, tmp, 0, known.length);
                 known = tmp;
             }
-            map.put(name, known);
+            if (validNameAndValue) {
```

Comment:
This condition can be placed above line 315 (new code), as we know, that `name` is `null` if `validNameAndValue` is false and we never put a value for the `null` key into the map.

Reply:
No sure, "known" variable is define at line 315, and test if null just after.
the block 
           " if (validNameAndValue) {
                map.put(name, known);
            }"
seems at the good place (line 325)

Reply:
We don't use the variable `known`, if `validNameAndValue` is false, so we can skip setting it. That is why, I suggest placing the condition before line 325.

Reply:
@FSchumacher I not sure that I understand... the if condition with validNameAndValue is for put in the map name / know values with avoid to do this if the name&value is null (special case "&=&").  so the if condition must be just above the "map.put(name, known);" in line 326. so I don't understand how put the if condition more higher in the code?

Reply:
If `validNameAndValue` is false, `name` is `null`.
If `validNameAndValue` is true, `name` is not `null`.
The only code path that leads to putting something into the map (and not return directly) is where `name` is not `null`.
Therefore we can skip all code, that is looking for values in the map, when `name` is `null`.
Therefore we can widen the block from `if (validNameAndValue)` to line 314.

Reply:
Maybe it is easier to use `continue` in case `paramSplit.length == 1`
Then we would have something like:
```java
String[] paramSplit = param.split("=");
if (paramSplit.length == 1) {
    continue; // skip in case, we have no key and value-pair
}
String name = decodeQuery(paramSplit[0]);
...
// no special handling for single `=` char needed from here
```

Reply:
@FSchumacher  no continue break the for loop and if params string contains value after the empty name/value like "&=&k2=v2", the k2 will be missing

Reply:
> If `validNameAndValue` is false, `name` is `null`. If `validNameAndValue` is true, `name` is not `null`. The only code path that leads to putting something into the map (and not return directly) is where `name` is not `null`. Therefore we can skip all code, that is looking for values in the map, when `name` is `null`. Therefore we can widen the block from `if (validNameAndValue)` to line 314.

With a params with no name but with one value, that will no works (like my test case "k1=v1&=value&k2=v2".

Reply:
You can take a look at https://github.com/FSchumacher/jmeter/commit/02f8ce8c66c3434b5e8e565dff3c7835aabb75c1 where I implemented the changes. The tests pass.
In case of `=value` we have `paramSplit.length==2`and thus `name==""` and `validNameAndValue==true`. That means, the code block inside the if condition will still be run. (Or alternatively the `continue` will work)